javassist.CannotCompileException

Hi,


I'm using version 0.89.1  for Android and I keep getting the above mentioned error. My Stack trace is written below

```
Error:17:35:13.605 [ERROR] [system.err] Note: Processing class Addon
17:35:13.605 [ERROR] [system.err] Note: Processing class Appointment
17:35:13.606 [ERROR] [system.err] Note: Processing class Area
17:35:13.606 [ERROR] [system.err] Note: Processing class Categories
17:35:13.606 [ERROR] [system.err] Note: Processing class Clinic
17:35:13.606 [ERROR] [system.err] Note: Processing class CompletionCategories
17:35:13.606 [ERROR] [system.err] Note: Processing class Country
17:35:13.606 [ERROR] [system.err] Note: Processing class Day
17:35:13.606 [ERROR] [system.err] Note: Processing class DayMealCategory
17:35:13.606 [ERROR] [system.err] Note: Processing class Doctor
17:35:13.606 [ERROR] [system.err] Note: Processing class Item
17:35:13.606 [ERROR] [system.err] Note: Processing class ItemsSelections
17:35:13.606 [ERROR] [system.err] Note: Processing class Meal
17:35:13.606 [ERROR] [system.err] Note: Processing class Order
17:35:13.606 [ERROR] [system.err] Note: Processing class Package
17:35:13.607 [ERROR] [system.err] Note: Processing class Paths
17:35:13.607 [ERROR] [system.err] Note: Processing class Province
17:35:13.607 [ERROR] [system.err] Note: Processing class Salt
17:35:13.607 [ERROR] [system.err] Note: Processing class SelectedDate
17:35:13.607 [ERROR] [system.err] Note: Processing class Selections
17:35:13.607 [ERROR] [system.err] Note: Processing class Settings
17:35:13.607 [ERROR] [system.err] Note: Processing class Sex
17:35:13.607 [ERROR] [system.err] Note: Processing class Time
17:35:13.607 [ERROR] [system.err] Note: Processing class User
17:35:13.607 [ERROR] [system.err] Note: Creating DefaultRealmModule
17:35:14.259 [ERROR] [system.err] Note: Some input files use unchecked or unsafe operations.
17:35:14.260 [ERROR] [system.err] Note: Recompile with -Xlint:unchecked for details.
17:35:14.576 [ERROR] [org.gradle.BuildExceptionReporter] 
17:35:14.576 [ERROR] [org.gradle.BuildExceptionReporter] FAILURE: Build failed with an exception.
17:35:14.576 [ERROR] [org.gradle.BuildExceptionReporter] 
17:35:14.576 [ERROR] [org.gradle.BuildExceptionReporter] * What went wrong:
17:35:14.576 [ERROR] [org.gradle.BuildExceptionReporter] Execution failed for task ':app:transformClassesWithRealmTransformerForDebug'.
17:35:14.576 [ERROR] [org.gradle.BuildExceptionReporter] > javassist.CannotCompileException: showMenu ()V in dietfix.com.mahmoudhaddad.dietfix.MenuActivity: failed to resolve types
17:35:14.576 [ERROR] [org.gradle.BuildExceptionReporter] 
17:35:14.576 [ERROR] [org.gradle.BuildExceptionReporter] * Exception is:
17:35:14.577 [ERROR] [org.gradle.BuildExceptionReporter] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:transformClassesWithRealmTransformerForDebug'.
17:35:14.577 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
17:35:14.577 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
17:35:14.577 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
17:35:14.577 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
17:35:14.577 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
17:35:14.577 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
17:35:14.578 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
17:35:14.578 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
17:35:14.578 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
17:35:14.578 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
17:35:14.578 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
17:35:14.578 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:66)
17:35:14.578 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
17:35:14.578 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:25)
17:35:14.578 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:110)
17:35:14.578 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
17:35:14.578 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
17:35:14.578 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
17:35:14.579 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
17:35:14.579 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
17:35:14.579 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
17:35:14.579 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
17:35:14.579 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.initialization.DefaultGradleLauncher$4.run(DefaultGradleLauncher.java:154)
17:35:14.579 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.internal.Factories$1.create(Factories.java:22)
17:35:14.579 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
17:35:14.579 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:52)
17:35:14.579 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:151)
17:35:14.579 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:32)
17:35:14.579 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:99)
17:35:14.579 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:93)
17:35:14.580 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
17:35:14.580 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:62)
17:35:14.580 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:93)
17:35:14.580 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:82)
17:35:14.580 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:94)
17:35:14.580 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.tooling.internal.provider.runner.BuildModelActionRunner.run(BuildModelActionRunner.java:46)
17:35:14.580 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
17:35:14.580 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.tooling.internal.provider.runner.SubscribableBuildActionRunner.run(SubscribableBuildActionRunner.java:58)
17:35:14.580 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
17:35:14.580 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:43)
17:35:14.580 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:28)
17:35:14.580 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:78)
17:35:14.580 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:48)
17:35:14.581 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:52)
17:35:14.581 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
17:35:14.581 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
17:35:14.581 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:37)
17:35:14.581 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
17:35:14.581 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
17:35:14.581 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
17:35:14.581 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
17:35:14.581 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
17:35:14.581 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
17:35:14.581 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
17:35:14.581 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.util.Swapper.swap(Swapper.java:38)
17:35:14.581 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
17:35:14.582 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
17:35:14.582 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.health.DaemonHealthTracker.execute(DaemonHealthTracker.java:47)
17:35:14.582 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
17:35:14.582 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:66)
17:35:14.582 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
17:35:14.582 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
17:35:14.582 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:72)
17:35:14.582 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
17:35:14.582 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
17:35:14.582 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.health.HintGCAfterBuild.execute(HintGCAfterBuild.java:41)
17:35:14.582 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
17:35:14.582 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
17:35:14.582 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:246)
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] Caused by: java.lang.RuntimeException: javassist.CannotCompileException: showMenu ()V in dietfix.com.mahmoudhaddad.dietfix.MenuActivity: failed to resolve types
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] 	at com.android.builder.profile.Recorder$Block.handleException(Recorder.java:54)
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] 	at com.android.builder.profile.ThreadRecorder$1.record(ThreadRecorder.java:57)
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] 	at com.android.builder.profile.ThreadRecorder$1.record(ThreadRecorder.java:47)
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] 	at com.android.build.gradle.internal.pipeline.TransformTask.transform(TransformTask.java:173)
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$IncrementalTaskAction.doExecute(AnnotationProcessingTaskFactory.java:244)
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:220)
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$IncrementalTaskAction.execute(AnnotationProcessingTaskFactory.java:231)
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:209)
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] 	... 70 more
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] Caused by: javassist.CannotCompileException: showMenu ()V in dietfix.com.mahmoudhaddad.dietfix.MenuActivity: failed to resolve types
17:35:14.583 [ERROR] [org.gradle.BuildExceptionReporter] 	at javassist.expr.ExprEditor.doit(ExprEditor.java:117)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at javassist.CtBehavior.instrument(CtBehavior.java:712)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at javassist.CtBehavior$instrument.call(Unknown Source)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at io.realm.transformer.BytecodeModifier$_useRealmAccessors_closure2.doCall(BytecodeModifier.groovy:73)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at io.realm.transformer.BytecodeModifier.useRealmAccessors(BytecodeModifier.groovy:61)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at io.realm.transformer.BytecodeModifier$useRealmAccessors$2.call(Unknown Source)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at io.realm.transformer.RealmTransformer$_transform_closure11.doCall(RealmTransformer.groovy:131)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at io.realm.transformer.RealmTransformer.transform(RealmTransformer.groovy:128)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at com.android.build.api.transform.Transform.transform(Transform.java:308)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at com.android.build.gradle.internal.pipeline.TransformTask$3.call(TransformTask.java:178)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at com.android.build.gradle.internal.pipeline.TransformTask$3.call(TransformTask.java:174)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at com.android.builder.profile.ThreadRecorder$1.record(ThreadRecorder.java:55)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	... 79 more
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] Caused by: javassist.bytecode.BadBytecode: showMenu ()V in dietfix.com.mahmoudhaddad.dietfix.MenuActivity: failed to resolve types
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at javassist.bytecode.stackmap.MapMaker.make(MapMaker.java:111)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at javassist.bytecode.MethodInfo.rebuildStackMap(MethodInfo.java:440)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at javassist.bytecode.MethodInfo.rebuildStackMapIf6(MethodInfo.java:422)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at javassist.expr.ExprEditor.doit(ExprEditor.java:113)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	... 90 more
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] Caused by: javassist.bytecode.BadBytecode: failed to resolve types
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at javassist.bytecode.stackmap.MapMaker.make(MapMaker.java:169)
17:35:14.584 [ERROR] [org.gradle.BuildExceptionReporter] 	at javassist.bytecode.stackmap.MapMaker.make(MapMaker.java:108)
17:35:14.585 [ERROR] [org.gradle.BuildExceptionReporter] 	... 93 more
17:35:14.585 [ERROR] [org.gradle.BuildExceptionReporter] Caused by: javassist.NotFoundException: org.json.JSONException
17:35:14.585 [ERROR] [org.gradle.BuildExceptionReporter] 	at javassist.ClassPool.get(ClassPool.java:452)
17:35:14.585 [ERROR] [org.gradle.BuildExceptionReporter] 	at javassist.bytecode.stackmap.TypeData$TypeVar.fixTypes2(TypeData.java:347)
17:35:14.585 [ERROR] [org.gradle.BuildExceptionReporter] 	at javassist.bytecode.stackmap.TypeData$TypeVar.fixTypes(TypeData.java:330)
17:35:14.585 [ERROR] [org.gradle.BuildExceptionReporter] 	at javassist.bytecode.stackmap.TypeData$TypeVar.dfs(TypeData.java:274)
17:35:14.585 [ERROR] [org.gradle.BuildExceptionReporter] 	at javassist.bytecode.stackmap.MapMaker.fixTypes(MapMaker.java:390)
17:35:14.585 [ERROR] [org.gradle.BuildExceptionReporter] 	at javassist.bytecode.stackmap.MapMaker.make(MapMaker.java:167)
17:35:14.585 [ERROR] [org.gradle.BuildExceptionReporter] 	... 94 more
17:35:14.585 [ERROR] [org.gradle.BuildExceptionReporter] 
```

`Caused by: javassist.NotFoundException: android.content.ActivityNotFoundException`
`ActivityNotFoundException` is being used in `openBookUsingIntent`. When replace the field accessor, javaassist cannot rebuild the stack map because of the `ActivityNotFoundException` cannot be found.

I think we need to find a way to add `mockable-android-xx.jar` into the class pool.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,4 +1,4 @@
-## 0.91.0
+## 0.90.1
 
 * Updated Realm Core to 0.100.2.
 
@@ -7,6 +7,7 @@
 * Opening a Realm while closing a Realm in another thread could lead to a race condition.
 * Automatic migration to the new file format could in rare circumstances lead to a crash.
 * Fixing a race condition that may occur when using Async API (#2724).
+* Fixed CannotCompileException when related class definition in android.jar cannot be found (#2703).
 
 ### Enhancements
 

--- a/gradle-plugin/src/main/groovy/io/realm/gradle/Realm.groovy
+++ b/gradle-plugin/src/main/groovy/io/realm/gradle/Realm.groovy
@@ -47,7 +47,7 @@ class Realm implements Plugin<Project> {
             project.plugins.apply(AndroidAptPlugin)
         }
 
-        project.android.registerTransform(new RealmTransformer())
+        project.android.registerTransform(new RealmTransformer(project))
         project.repositories.add(project.getRepositories().jcenter())
         project.dependencies.add("compile", "io.realm:realm-android-library:${Version.VERSION}")
         project.dependencies.add("compile", "io.realm:realm-annotations:${Version.VERSION}")

--- a/realm-transformer/src/main/groovy/io/realm/transformer/RealmTransformer.groovy
+++ b/realm-transformer/src/main/groovy/io/realm/transformer/RealmTransformer.groovy
@@ -27,6 +27,7 @@ import io.realm.annotations.RealmClass
 import javassist.ClassPool
 import javassist.CtClass
 import javassist.LoaderClassPath
+import org.gradle.api.Project
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 
@@ -43,6 +44,11 @@ import static com.android.build.api.transform.QualifiedContent.*
 class RealmTransformer extends Transform {
 
     private Logger logger = LoggerFactory.getLogger('realm-logger')
+    private Project project
+
+    public RealmTransformer(Project project) {
+        this.project = project
+    }
 
     @Override
     String getName() {
@@ -84,6 +90,9 @@ class RealmTransformer extends Transform {
 
         // Create and populate the Javassist class pool
         ClassPool classPool = createClassPool(inputs, referencedInputs)
+        // Append android.jar to class pool. We don't need the class names of them but only the class in the pool for
+        // javassist. See https://github.com/realm/realm-java/issues/2703.
+        addBootClassesToClassPool(classPool)
 
         logger.info "ClassPool contains Realm classes: ${classPool.getOrNull('io.realm.RealmList') != null}"
 
@@ -278,4 +287,19 @@ class RealmTransformer extends Transform {
         return merged;
     }
 
+    // There is no official way to get the path to android.jar for transform.
+    // See https://code.google.com/p/android/issues/detail?id=209426
+    private void addBootClassesToClassPool(ClassPool classPool) {
+        try {
+            project.android.bootClasspath.each {
+                String path = it.absolutePath
+                logger.info "Add boot class " + path + " to class pool."
+                classPool.appendClassPath(path)
+            }
+        } catch (Exception e) {
+            // Just log it. It might not impact the transforming if the method which needs to be transformer doesn't
+            // contain classes from android.jar.
+            logger.info("Cannot get bootClasspath caused by:", e)
+        }
+    }
 }

